connector/spanmetricsconnector/connector.go

```diff
@@ -334,13 +340,31 @@ func (p *connectorImp) getOrCreateResourceMetrics(attr pcommon.Map) *resourceMet
 	return v
 }
 
+// contains checks if string slice contains a string value
+func contains(elements []string, value string) bool {
```

Comment:
Actually I do not think that it may cause problems in future.
It's a basic function that helps to check if element is  in array. It was copy-pasted from other part of collector-contrib. Unfortunately I did not find any common place for such functions in this repo.

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/8dc41a8582dae9680e49c8fa997fac4fc1487be7/receiver/snmpreceiver/config.go#L575
https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/8dc41a8582dae9680e49c8fa997fac4fc1487be7/pkg/translator/prometheus/normalize_name.go#L292

Reply:
Thanks for giving a bit of background. I think the difference here from those examples is that we're checking if a static item (the attribute key) is present in a static set (a list of attribute keys defined in a config) in the hot path of the component. In the SNMP receiver that check is only done once at component startup and in the Prometheus translator the data in the slice is generated from incoming data.

That said, for a slice of four items the performance difference would probably be negligible, I just wanted to ask about it.